app/uk/gov/hmrc/agentsubscriptionfrontend/service/HostnameWhiteListService.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentsubscriptionfrontend.service
+
+import java.net.URL
+import javax.inject.{Inject, Singleton}
+
+import uk.gov.hmrc.agentsubscriptionfrontend.config.AppConfig
+import uk.gov.hmrc.agentsubscriptionfrontend.connectors.SsoConnector
+import uk.gov.hmrc.play.binders.ContinueUrl
+import uk.gov.hmrc.play.http.HeaderCarrier
+
+import scala.concurrent.Future
+import scala.util.{Success, Try}
+
+@Singleton
+class HostnameWhiteListService @Inject()(appConfig: AppConfig, ssoConnector: SsoConnector) {
```

Comment:
Artur mentions that it actually only checks the host portion of the URL anyway, so feel free to ignore this one, the name is fine